src/metabase/models/collection.clj

```diff
@@ -1221,29 +1227,38 @@
                                :allowed-namespaces   allowed-namespaces
                                :collection-namespace collection-namespace})))))))
 
-(defn- annotate-collections
+(defn annotate-collections
```

Comment:
What's the main change here? Is it just changing `:below` to reflect the entire subtree instead of just one level down? Maybe that can be made clearer in the doc string since the code is a bit dense

Reply:
The difference is more about generality. The old code took a map of `{:dataset set-of-collections-that-contain-datasets :card set-of-collections-that-contain-cards}` and annotated by setting `:here` and `:below` to a subset of `#{:card :dataset}`. In other words it just annotated `:card` and `:dataset`.

The new code takes a map of `{[arbitrary-entity] set-of-collections-that-contain-that-entity-type}` and annotates by setting `:here` and `:below` accordingly. I used it here to add `:collection` and `:dashboard` to the possible set in `:here`/`:below`.